extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/BasicSecurityDruidModule.java

```diff
@@ -168,6 +214,29 @@ public static BasicAuthorizerCacheNotifier createAuthorizerCacheNotifier(final I
     );
   }
 
+  /**
+   * Returns the instance provided either by a config property or coordinator-run class or default class.
+   * The order of check corresponds to the order of method params.
+   */
+  private static <T> T getInstance(
+      Injector injector,
+      String configClassName,
+      Class<? extends T> classRunByCoordinator,
+      Class<? extends T> defaultClass
+  ) throws ClassNotFoundException
+  {
+    if (configClassName != null) {
+      // ClassCastException is thrown in case of a mismatch, configuration fix is required.
+      @SuppressWarnings("unchecked")
+      final T instance = (T) injector.getInstance(Class.forName(configClassName));
```

Comment:
I think we exposing too much internals in configs propagating the class names for each sub-component. In fact, we would like a switch like `...authorization.management.enabled=true/false` which can effectively select required classes for us.

Reply:
This will require refactoring current DI of the module

Reply:
That works for testing, but it's awkward for production.

Reply:
Agree, this might be improved

Reply:
Let's add it to the issue we create for https://github.com/metamx/druid/pull/73#discussion_r298938974